app/apollo/resolvers/channel.js

```diff
@@ -694,12 +712,30 @@ const channelResolvers = {
 
       const user = whoIs(me);
 
-      try{
-        logger.info({req_id, user, org_id, uuid }, `${queryName} validating`);
+      try {
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating`);
 
         validateString( 'org_id', org_id );
         validateString( 'uuid', uuid );
 
+        // Find version (avoid using deprecated/ignored `versions` attribute on the channel)
+        const version = await models.DeployableVersion.findOne( { uuid, org_id } );
+
+        const channel = await models.Channel.findOne( { uuid: version.channel_id, org_id } );
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - channel found: ${!!channel}`);
+
+        const identifiers = channel ? [uuid, channel.name] : [uuid];
+        await validAuth(me, org_id, ACTIONS.MANAGEVERSION, TYPES.CHANNEL, queryName, context, identifiers);
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - authorized`);
+
+        if( !version ){
+          throw new NotFoundError( context.req.t( 'Version uuid "{{version_uuid}}" not found.', { 'version_uuid': uuid } ), context );
+        }
+
+        if( !channel ){
+          throw new NotFoundError( context.req.t( 'Channel uuid "{{channel_uuid}}" not found.', { 'channel_uuid': version.channel_id } ), context );
+        }
+
```

Comment:
Why were existence and auth checks moved earlier in the flow, before `EXPERIMENTAL_GITOPS_ALT` check and other simpler checks that can be done without expensive db/auth checks?  The order shouldn't be changed unless there was a specific reason?

Reply:
Reasoning being our conversation about checking auth before throwing validation errors that could give more information in those errors messages than should be allowed before auth is validated.

Reply:
In this case, the expensive auth checks should occur after 
- `if( !description && !remote ){` -- this is validating input without querying and exposing any information from the backend
- `if( subscriptions ) {` -- this is validating input without querying and exposing any information from the backend

Reply:
Unresolving, not addressed

Reply:
I moved expensive auth checks to after if( !description && !remote ){ and if( subscriptions ) {

Reply:
I think the fault was on my end -- refreshing a diff between specific commits rather than getting the overall diff for the PR fresh.
Resolved.